.travis.yml

```diff
@@ -17,12 +17,10 @@ jdk:
   - oraclejdk8
 
 install:
-  - mvn clean install -DskipTests -Dassembly.skipAssembly=true > /dev/null
+  - bash .travis_install.sh
 
 script:
-  - cd $PINOT_MODULE
-  - cp ../.travis_cmd.sh .
-  - bash .travis_cmd.sh
+  - bash .travis_test.sh
 
 after_success:
   - bash <(curl -s https://codecov.io/bash)
```

Comment:
There is a difference. If you `git clone 1234567` by definition either hash 1234567 exists and has the content that you expect, unless someone has created malicious content that happens to have the exact same SHA-1 hash. This is unlikely, but not impossible (see https://www.schneier.com/blog/archives/2012/10/when_will_we_se.html).

Pulling it from an external server adds the risk that the server is compromised somehow and serves a completely different version; checking out a known revision eliminates that risk as we can vet the contents of the script ahead of time and, barring a hash collision attack, that someone inserts code in that script.

I'd replace the bash/curl call by:
```
git clone https://github.com/codecov/codecov-bash.git
git checkout d3157145db519e2603b870e9596b9086a7983ac1 && so codecov/codecov
```